pyproject.toml

```diff
@@ -23,8 +23,8 @@ dependencies = [
     "importlib_resources>=5.0; python_version < '3.10'",
     "qiskit>=1.0.0",
     "joblib>=1.3.2",
-    "numpy>=1.26.4; python_version > '3.11'",
-    "numpy>=1.17.0; python_version <= '3.11'",
+    "numpy>=1.26.4,<2.0; python_version > '3.11'",
+    "numpy>=1.17.0,<2.0; python_version <= '3.11'",
```

Comment:
Maybe add comments to these dependencies as well, so that we do not forget that this is only temporary and also communicate that to people using our tool.
Something like:
```suggestion
    "numpy>=1.26.4,<2.0; python_version > '3.11'", # some of our dependencies are not yet compatible with numpy 2.0
    "numpy>=1.17.0,<2.0; python_version <= '3.11'", # some of our dependencies are not yet compatible with numpy 2.0
```